Mathlib/Tactic/Algebraize.lean

```diff
@@ -0,0 +1,264 @@
+/-
+Copyright (c) 2024 Calle Sönne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin, Nick Kuhn, Arend Mellendijk, Christian Merten, Calle Sönne, Adam Topaz
+-/
+
+import Mathlib.Algebra.Algebra.Tower
+
+/-!
+
+## Algebraize tactic
+
+This file defines the `algebraize` tactic. The basic functionality of this tactic is to
+automatically add `Algebra` instances given `RingHom`s. For example, `algebraize f g` where
+`f : A →+* B` and `g : B →+* C` are `RinHom`'s, will add the instances `Algebra A B` and
+`Algebra B C` corresponding to these `RingHom`s.
+
+# Further functionality
+
+When given a composition of `RingHom`'s, e.g. `algebraize g.comp f`, the tactic will also try to
+add the instance `IsScalarTower A B C` if possible.
+
+After having added suitable `Algebra` and `IsScalarTower` instances, the tactic will search through
+the local context for `RingHom` properties that can be converted to properties of the corresponding
+`Algebra`. For example, given `f : A →+* B` and `hf : f.FiniteType`, then `algebraize f` will add
+the instance `Algebra A B` and the corresponding property `Algebra.FiniteType A B`. The tactic knows
+which `RingHom` properties have a corresponding `Algebra` property through the `algebraize`
+attribute.
+
+# Algebraize attribute
+
+The `algebraize` attribute is used to tag `RingHom` properties that can be converted to `Algebra`
+properties. Using an (optional) parameter, it will also generate a `Name` of a declaration which
+will help the tactic to access the corresponding `Algebra` property.
+
+There are two cases for what declaration corresponding to this `Name` can be.
+1. An inductive type (i.e. the `Algebra` property itself), in this case it is assumed that the
+`RingHom` and the `Algebra` property are definitionally the same, and the tactic will construct the
+`Algebra` property by giving the `RingHom` property as a term.
+
+2. A constructor for the `Algebra` property. In this case it is assumed that the `RingHom` property
+is the last argument of the constructor, and that no other explicit argument is needed. The tactic
+then constructs the `Algebra` property by applying the constructor to the `RingHom` property.
+
+Finally, if no argument is provided to the `algebraize` attribute, it is assumed that the tagged
+declaration has name `RingHom.Property` and that the corresponding `Algebra` property has name
+`Algebra.Property`. The attribute then returns `Algebra.Property` (so assume case 1 above).
```

Comment:
> [...] Using an (optional) parameter, it will also generate a `Name` of a declaration which
> will help the tactic to access the corresponding `Algebra` property.
> 
> There are two cases for what declaration corresponding to this `Name` can be.
> 1. An inductive type (i.e. the `Algebra` property itself), in this case it is assumed that the
> `RingHom` and the `Algebra` property are definitionally the same, and the tactic will construct the
> `Algebra` property by giving the `RingHom` property as a term.
> 
> 2. A constructor for the `Algebra` property. In this case it is assumed that the `RingHom` property
> is the last argument of the constructor, and that no other explicit argument is needed. The tactic
> then constructs the `Algebra` property by applying the constructor to the `RingHom` property.
> 
> Finally, if no argument is provided to the `algebraize` attribute, it is assumed that the tagged
> declaration has name `RingHom.Property` and that the corresponding `Algebra` property has name
> `Algebra.Property`. The attribute then returns `Algebra.Property` (so assume case 1 above).


I find this a bit hard to read, especially the first sentense.

Maybe

```suggestion
The `algebraize` attribute is used to tag `RingHom` properties that can be converted to `Algebra`
properties. It assumes that the tagged
declaration has a name of the form `RingHom.Property` and that the corresponding `Algebra` property has the name `Algebra.Property`.

If not, the `Name` of the corresponding algebra property can be provided as optional argument. The specified declaration should be one of the following:

1. An inductive type (i.e. the `Algebra` property itself), in this case it is assumed that the
`RingHom` and the `Algebra` property are definitionally the same, and the tactic will construct the
`Algebra` property by giving the `RingHom` property as a term.

2. A constructor for the `Algebra` property. In this case it is assumed that the `RingHom` property
is the last argument of the constructor, and that no other explicit argument is needed. The tactic
then constructs the `Algebra` property by applying the constructor to the `RingHom` property.
```

Reply:
This is great, thanks a lot!